core/tests/protractor_utils/BlogDashboardPage.js

```diff
@@ -0,0 +1,319 @@
+
+  this.saveBlogPostAsDraft = async function(
+      blogPostTitle, richTextInstructions) {
+    await action.sendKeys(
+      'New blog post title field', BlogPostTitleFieldElement, blogPostTitle);
+    await this.setContent(richTextInstructions);
+    await waitFor.visibilityOf(
+      blogPostContentDisplay, 'Blog Post content not showing up');
+    await action.click('Save as draft Button', saveBlogPostAsDraftButton);
+    await waitFor.visibilityOfSuccessToast('Blog Post Saved Successfully.');
+  };
+
+  this.publishBlogPost = async function(
+      blogPostTitle, richTextInstructions, tags) {
+    await action.sendKeys(
+      'New blog post title field', BlogPostTitleFieldElement, blogPostTitle);
+    await this.selectTags(tags);
+    await this.setContent(richTextInstructions);
+    await waitFor.visibilityOf(
+      blogPostContentDisplay, 'Blog Post content not showing up');
+    await this.setThumbnailImage();
+    await action.click('Publish Blog Post', publishBlogPostButton);
+    await action.click(
+      'Confirm Publish Blog Post button', confirmButton);
+    await waitFor.visibilityOfSuccessToast(
+      'Blog Post Saved and Published Succesfully.');
+  };
+
+  this.selectTags = async function(tags) {
+    for (let i = 0; i < tags.length; i++) {
+      await action.click(
+        'select blog post tag', blogPostTags.get(tags[i]));
+    }
+  };
+
+  this.navigateToPublishTab = async function() {
+    await waitFor.pageToFullyLoad();
+    await waitFor.visibilityOf(
+      publishTab, 'Unable to find publish tab button');
+    await action.click('Publish Tab', publishTab);
+  };
+
+  this.navigateToDraftsTab = async function() {
+    await waitFor.pageToFullyLoad();
+    await waitFor.visibilityOf(
+      draftTab, 'Unable to find drafts tab button');
+    await action.click('Draft Tab', draftTab);
+  };
+
+  this.expectNumberOfDraftsBlogPostsToBe = async function(number) {
+    var isPresent = await blogDashboardIntroMessage.isPresent();
+    if (!isPresent) {
+      await this.waitForDraftsBlogPostsToLoad();
+    }
+    expect(await draftBlogPostTiles.count()).toBe(number);
+  };
+
+  this.expectNumberOfPublishedBlogPostsToBe = async function(number) {
+    var isPresent = await blogDashboardIntroMessage.isPresent();
+    if (!isPresent) {
+      await this.waitForPublishedBlogPostsToLoad();
+    }
+    expect(await publishedBlogPostTiles.count()).toBe(number);
+  };
+
+  this.expectNumberOfBlogPostsRowsToBe = async function(number) {
+    var isPresent = await blogDashboardIntroMessage.isPresent();
+    if (!isPresent) {
+      await this.waitForDraftsBlogPostsToLoad();
+    }
+    expect(await blogPostListItems.count()).toBe(number);
+  };
+
+  this.navigateToBlogPostEditorWithIndexFromList = async function(index) {
+    await action.click('blog post row', blogPostListItems.get(index));
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.getBlogPostCount = async function() {
+    return await BlogPostListItems.count();
```

Comment:
What if the items haven't loaded yet?

Reply:
Removed the complete function. It is not required.